src/asyncProfiler.cpp

```diff
@@ -150,6 +150,16 @@ void Profiler::storeMethod(jmethodID method) {
 }
 
 void Profiler::recordSample(void* ucontext) {
+    // Check deadline
+    if (time(NULL) > _deadline) {
+        char error[] = "Disabling profiler due to deadline\n";
+        write(STDERR_FILENO, error, sizeof(error));
```

Comment:
What can I do with the return value in this case? It is an error reporting in signal handler.
BTW, no warning in case of Mac OS:
```
$ make
mkdir -p build
g++ -O2 -I/Library/Java/JavaVirtualMachines/jdk1.8.0_111.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk1.8.0_111.jdk/Contents/Home/include/darwin -fPIC -shared -o build/libasyncProfiler.so src/*.cpp
```
Or Linux:
```
$ make
mkdir -p build
g++ -O2 -I/usr/java/jdk180_64_60/include -I/usr/java/jdk180_64_60/include/linux -fPIC -shared -o build/libasyncProfiler.so src/*.cpp
```